### DISCUSSION TITLE
Need way to link a `ProjectV2` to a `Repository`

### DISCUSSION THREAD

**Discussion by heaths** [community/community#16168] [Projects and Issues] [answered]
There is currently a [`linkRepositoryToProject`](https://docs.github.com/graphql/reference/mutations#linkrepositorytoproject) but nothing like that I can find to link a repo to a `ProjectV2`. The goal here is to reduce the number of queries we'd otherwise have to do. For example, if a `ProjectV2` is already linked with a `Repository`, we only need to query `repository { projectsV2 }`; otherwise, we'd also have to query `organization { projectsV2 }` and `user { projectsV2 }`. However, if custom UX atop the GraphQL support for Projects (beta) could react to "not found" errors it could check either the organization or user for a project number, get the ID, and link it once time instead of checking all three root objects every time.

**Comment by tvquizphd** [comment]
With `ProjectV2`, this continues to be [an issue](https://github.com/community/community/discussions/35073).
Edit: I've opened [an issue](https://github.com/octokit/graphql-schema/issues/641) in the graphql-schema repo.

**Comment by heaths** [comment, marked as answer]
This is now possible with the mutation [`linkProjectV2ToRepository`](https://docs.github.com/en/graphql/reference/mutations#linkprojectv2torepository). I've got an example of usage in https://github.com/heaths/gh-projects/pull/40
